src/bson.c

```diff
@@ -605,21 +605,39 @@ static bool php_phongo_bson_visit_symbol(const bson_iter_t* iter, const char* ke
 	return false;
 } /* }}} */
 
-static void php_phongo_bson_new_javascript_from_javascript_and_scope(zval* object, const char* code, size_t code_len, const bson_t* scope TSRMLS_DC) /* {{{ */
+static bool php_phongo_bson_new_javascript_from_javascript_and_scope(zval* object, const char* code, size_t code_len, const bson_t* scope TSRMLS_DC) /* {{{ */
 {
 	php_phongo_javascript_t* intern;
 
+	if (scope) {
+		php_phongo_bson_state state;
+
+		PHONGO_BSON_INIT_STATE(state);
+
+		if (!php_phongo_bson_to_zval_ex(bson_get_data(scope), scope->len, &state)) {
+			zval_ptr_dtor(&state.zchild);
+			php_phongo_bson_state_dtor(&state);
```

Comment:
Is this redundant? I think it relates to https://github.com/mongodb/mongo-php-driver/pull/1034#discussion_r338642376.